custom/icds/view_utils.py

```diff
@@ -21,5 +24,16 @@ def _inner(request, domain, *args, **kwargs):
     return _inner
 
 
+def check_edit_access_for_domain(fn):
```

Comment:
we could not use `check_data_interfaces_blocked_for_domain`?

Reply:
We can, but we were not getting a meaningful message on UI. 
It was showing "Something went wrong" message.
So, I created a new decorator which informs about the reason why you can't make changes.

Reply:
I'd go out on a limb and say that that is okay i.e the "Something went wrong" message. Since the links are already hidden, someone would be able to send this request only if they do something fishy, and we should not worry about adding more code for their convinience.

I'd suggest using the available method only, which is the same as this new method with one line difference. If you still need a better message, you can may be possibly check if the request needs a JSON response and return that when needed which might be useful for new views that we need to block in future? And I think the UI should have made use of the `no_permission` repsonse, may be it does handle error messages well.
But in short, I don't see a real worth for adding the new method.